src/nnet3/nnet-training.cc

```diff
@@ -235,6 +235,45 @@ void ComputeObjectiveFunction(const GeneralMatrix &supervision,
         computer->AcceptOutputDeriv(output_name, &diff);
       break;
     }
+    case kLognormal: {
```

Comment:
I am not sure about whether I think it's the right thing to add kLogNormal (should be capital N) to the directly-supported objective functions.  That list of objective functions was never meant to be exhaustive, merely include some of the super-easy ones.  It's also valid to ignore the objective function type in the enum, and write your own code to evaluate the objective.  But doing it this way might be easier.  This code might benefit from being in its own function (to avoid cluttering the place where it goes), and have longer comments explaining what's going on.
I'm not deciding this now-- it's going to be part of the decision later on of whether to include this whole phone-duration-modeling feature.  One of the things that will affect this is whether or not it gives improvements on top of the chain models.